### PR TITLE
Add Client side Encryption

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

### New

* add  native module AdyenCSE with methods 
  * `encryptCard(card: Object, publicKey: String) -> Promise`
  * `encryptBin(bin: String, publicKey: String) -> Promise`

### CODE REVIEW

File: android/src/main/java/com/adyenreactnativesdk/cse/AdyenCSEModule.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            map.putString(NUMBER_KEY, encryptedCard.encryptedCardNumber)
+            map.putString(EXPIRY_MONTH_KEY, encryptedCard.encryptedExpiryMonth)
+            map.putString(EXPIRY_YEAR_KEY, encryptedCard.encryptedExpiryYear)
+            map.putString(CVV_KEY, encryptedCard.encryptedSecurityCode)
+            promise.resolve(map)
+        } catch (e: EncryptionException) {
+            promise.reject(ERROR_MESSAGE, e)
+        }
+    }
+
```
Comment (OscarSpruit) [inline android/src/main/java/com/adyenreactnativesdk/cse/AdyenCSEModule.kt:38]: The `CardEncrypter` also has the `public static String encrypt(...)` method to encrypt all data into a single string. Did you leave that out on purpose?

Comment (descorp) [reply android/src/main/java/com/adyenreactnativesdk/cse/AdyenCSEModule.kt:38]: AFAIK it is for "Classic" PAL integration that was deprecated. Unless it is requested by someone - I see no reason adding it.

**Review by OscarSpruit — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

File: example/src/CSE.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    try {
+      const encryptedCard = await AdyenCSE.encryptCard(
+        unencryptedCard,
+        publicKey
+      );
+      console.log(JSON.stringify(encryptedCard));
+    } catch (e) {
+      console.error(e);
+    }
+  };
```
Comment (ribeiroguilherme) [inline example/src/CSE.js:27]: You can wrap the declared functions in `useCallback` hook - this way, they aren't recreated in memory if the Component re-renders.

```jsx
const tryEncryptCard = useCallback(async () => {
...
}. []);
```

File: example/src/CSE.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { Button, SafeAreaView } from 'react-native';
+import { NativeModules } from 'react-native';
+import { ENVIRONMENT } from './Configuration';
+import Styles from './Styles';
+
+const { AdyenCSE } = NativeModules;
+
+const CseView = () => {
+  const tryEncryptCard = async () => {
+    let unencryptedCard = {
```
Comment (ribeiroguilherme) [inline example/src/CSE.js:11]: Should be defined as `const` instead of `let`, as there is no need to mutate the object

File: example/src/CSE.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      number: '5454 5454 5454 5454',
+      expiryMonth: '03',
+      expiryYear: '2030',
+      cvv: '737',
+    };
+    let publicKey = ENVIRONMENT.publicKey;
```
Comment (ribeiroguilherme) [inline example/src/CSE.js:17]: Perhaps this publicKey could be moved to the top of the file after the imports, as it is a constant value used in this Component. You don't have to define it locally in both functions

```jsx
...
import { ENVIRONMENT } from './Configuration';
import Styles from './Styles';

const PUBLIC_KEY = ENVIRONMENT.publicKey;
...
```

**Review by ribeiroguilherme — APPROVED** [review]

File: example/src/CSE.js (hunk 2 above)
Comment (descorp) [reply example/src/CSE.js:27]: This will probably change (or maybe even removed).
I'll address this in next PR

File: example/src/CSE.js (hunk 3 above)
Comment (descorp) [reply example/src/CSE.js:11]: This will probably change (or maybe even removed).
I'll address this in next PR

File: example/src/CSE.js (hunk 4 above)
Comment (descorp) [reply example/src/CSE.js:17]: This will probably change (or maybe even removed).
I'll address this in next PR

**Pull request merged**